central/namespace/datastore/datastore.go

```diff
@@ -48,6 +50,14 @@ func New(nsStore store.Store, deploymentDataStore deploymentDataStore.DataStore,
 	}
 }
 
+// Constructor of the namespace storage takes a Store object as an argument,
+// but it's an internal type which could not be constructed outside. Such
+// approach limits the ways how namespace storage could be instantiated to only
+// the singleton. This method allows to avoid the limitation.
+func NewStorage(db pg.DB) store.Store {
```

Comment:
I think it would be nice if we could gate this somehow to limit its use.  There are very few circumstances where it makes sense to use this as the application itself needs the full capability of the datastore due to things being intertwined with the cache and other things.  So if someone accidentally used this within ACS we may have some unintended consequences

Reply:
The fact that the storage framework is impossible to use separately from the app is what makes experimenting with it (and and the end of the day, flexibility and clear understanding) harder. So I would prefer to not gate it, but rather similar to the other PR add a warning, which will be fired from this function, stating it's for testing purposes only. How does it sound?

Reply:
I did move the store from `central/namespace` to `central/namespace/datastore/internal` a few months ago, as I did not think it was making sense anymore to have it accessible out of the datastore layer.
One way of gating things would be to move it back, and add a roxvet rule (in validateimports) with specified exceptions to ensure the store is used in controlled manner out of the storage framework.